docs/source/conf.py

```diff
@@ -127,6 +165,88 @@
     "exclude-members": "__weakref__",
 }
 
+# Mock imports for dependencies that aren't available during doc building
```

Comment:
just for my understanding, once we have wheels that work in CI then we can go ahead and remove all of this right?

Reply:
I think it might be problematic for a local installation that doesn't have all these deps more than the CI since we are installing all this. Also, just for API docs these import might not be necessarily needed. Since I'm only adding doc build changes and sphinx gallery in this PR, I will delete this part for now.